docs/source/benchmarks.rst

```diff
@@ -28,31 +28,31 @@ Benchmarking invariant at __init__:
 =========================  ============  ==============  =======================
 Case                         Total time    Time per run    Relative time per run
 =========================  ============  ==============  =======================
-`ClassWithIcontract`             1.33 s         1.33 μs                     100%
-`ClassWithDpcontracts`           0.45 s         0.45 μs                      34%
-`ClassWithDeal`                  2.62 s         2.62 μs                     197%
-`ClassWithInlineContract`        0.27 s         0.27 μs                      20%
+`ClassWithIcontract`             1.45 s         1.45 μs                     100%
+`ClassWithDpcontracts`           0.48 s         0.48 μs                      33%
+`ClassWithDeal`                  1.73 s         1.73 μs                     119%
+`ClassWithInlineContract`        0.28 s         0.28 μs                      19%
 =========================  ============  ==============  =======================
 
 Benchmarking invariant at a function:
 
 =========================  ============  ==============  =======================
 Case                         Total time    Time per run    Relative time per run
 =========================  ============  ==============  =======================
-`ClassWithIcontract`             1.98 s         1.98 μs                     100%
-`ClassWithDpcontracts`           0.46 s         0.46 μs                      23%
-`ClassWithDeal`                  7.13 s         7.13 μs                     360%
-`ClassWithInlineContract`        0.23 s         0.23 μs                      12%
+`ClassWithIcontract`             2.04 s         2.04 μs                     100%
+`ClassWithDpcontracts`           0.49 s         0.49 μs                      24%
+`ClassWithDeal`                  4.67 s         4.67 μs                     230%
```

Comment:
Cool. Do you check invariants for setattr? I do and I think I should make it configurable, it's quite expensive. I assume that's where I lose most of the time in the test.

Reply:
Partially, the invariants are checked only before and after the method calls. So if you have a setter, the invariant will be checked, but otherwise not.